test/langtools/jdk/javadoc/doclet/testMethodCommentAlgorithm/TestMethodCommentsAlgorithm.java

```diff
@@ -0,0 +1,459 @@
+                %s
+                    @Override boolean equals(Object obj);
+                }
+                """.formatted(generateDocComment(4, i)), """
+                package x;
+                public interface T5 {
+                %s
+                    @Override boolean equals(Object obj);
+                }
+                """.formatted(generateDocComment(5, i)), """
+                package x;
+                public interface T6 {
+                %s
+                    @Override boolean equals(Object obj);
+                }
+                """.formatted(generateDocComment(6, i)), """
+                module mymodule { }
+                """);
+
+        createPatchedJavaLangObject(javaBase.resolve("share").resolve("classes").toAbsolutePath(),
+                Files.createDirectories(src.resolve("java.base")).toAbsolutePath(),
+                generateDocComment(7, i, false));
+
+        javadoc("-d", base.resolve("out-" + i).toAbsolutePath().toString(),
+                "-tag", "apiNote:a:API Note:",
+                "-tag", "implSpec:a:Implementation Requirements:",
+                "-tag", "implNote:a:Implementation Note:",
+                "--patch-module", "java.base=" + src.resolve("java.base").toAbsolutePath().toString(),
+                "--module-source-path", src.toAbsolutePath().toString(),
+                "mymodule/x");
+
+        checkExit(Exit.OK);
+    }
+
+    private static String generateDocComment(int index, int run) {
+        return generateDocComment(index, run, true);
+    }
+
+    /*
+     * Provides a doc comment for an override of Object.equals in a type with
+     * the specified index for the specified run.
+     */
+    private static String generateDocComment(int index, int run, boolean includeCommentMarkers) {
+        if (index > run) {
+            String s = """
+                    T%s: main description
+                    *
+                    * @param obj T%1$s: parameter description
+                    * @return T%1$s: return description""";
+            if (includeCommentMarkers)
+                s = "/**\n* " + s + "\n*/";
+            return s.formatted(index).indent(4);
+        } else {
+            return "";
+        }
+    }
+
+    /*
+     * Tests that the documentation search order is as shown:
+     *
+     *    (3) (4)
+     *     ^   ^
+     *      \ /
+     *      (2) (5)
+     *       ^   ^
+     *        \ /
+     *        (1)
+     *         |
+     *         v
+     *        [6]
+     *         *
+     */
+    @Test
+    public void testInterfaceHierarchy(Path base) throws Exception {
+        Path p = Path.of(System.getProperty("test.src", ".")).toAbsolutePath();
+        while (!Files.exists(p.resolve("TEST.ROOT"))) {
+            p = p.getParent();
+            if (p == null) {
+                throw new SkippedException("can't find TEST.ROOT");
+            }
+        }
+        System.err.println("Test suite root: " + p);
```

Comment:
You might want to use `JavadocTester.out` instead of `System.err`

Reply:
Fixed in:  cccffa660a8.